HW_13/src/HW_13/TypeScript.ts

```diff
@@ -0,0 +1,122 @@
+// Вам необхідно написати додаток Todo list. У списку нотаток повинні бути методи для додавання нового запису, 
+// видалення, редагування та отримання повної інформації про нотатку за ідентифікатором, а так само отримання списку всіх нотаток. 
+// Крім цього, у користувача має бути можливість позначити нотаток, як виконаний, і отримання інформації про те, 
+// скільки всього нотаток у списку і скільки залишилося невиконаними. Нотатки не повинні бути порожніми.
+
+// Кожний нотаток має назву, зміст, дату створення і редагування та статус. Нотатки бувають двох типів. 
+// Дефолтні та такі, які вимагають підтвердження при ридагуванні.
+// Окремо необхідно розширити поведінку списку та додати можливість пошуку нотатка за ім'ям або змістом.
+// Також окремо необхідно розширити список можливістю сортування нотаток за статусом або часом створення.
+
+type idNote = number;
+
+interface INote {
+    title: string;
+    content: string;
+    createdAt: Date | null;
+    updatedAt: Date | null;
+    id: idNote;
+    isCompleted: boolean;
+}
+
+type NoteUpdated = Partial<Pick<INote, 'title' | 'content'>>;
+
+interface ITodoList {
+    addNote(title: string, content: string): void;
+    deleteNote(id: idNote): void;
+    updateNote(id: idNote, payload: NoteUpdated): INote;
+    getNoteById(id: idNote) : INote | undefined;
+    getNotes(): INote[];
+    findByIndex(id: idNote): number;
+    getUncompletedNotes(): INote[];
+    searchNotes(query: string): INote[];
+    sortNotesByStatus(): INote[];
+    sortNotesByCreationTime(): INote[];
+}
+
+class TodoList implements ITodoList {
+    public notes: INote[] = [];
+
+    public addNote(title: string, content: string): void {
+        const note: INote = new Note(title, content);
+        this.notes.push(note);
+    }
+
+    public deleteNote(id: idNote): void {
+        const index = this.findByIndex(id);
+        if (index !== -1) {
+            this.notes.splice(index, 1);
+        } else {
+            throw new Error('Нотатка не знайдена');
+        }
+    }
+
+    public updateNote(id: idNote, payload: NoteUpdated): INote {
+        const note = this.getNoteById(id);
+        if (note) {
+            note.title = payload.title || note.title;
+            note.content = payload.content || note.content;
+            note.updatedAt = new Date();
```

Comment:
Just wanted to let you know that there's a better way to update the Note class. By using the update method of Note class instead, you'll be following the Single Responsibility principle and keeping things running smoothly. Plus, it'll help make your code more organized and efficient!